enarx-pr-assign

```diff
@@ -8,30 +8,56 @@ class Override(Exception):
     def __init__(self, responsible):
         self.responsible = responsible
 
-def merge_events_and_reviews(pr):
+def merge_events_and_reviews(repo, pr):
     "Gets a summary of all events on a PR by merging issue events and reviews."
-    def inner(pr):
-        for e in pr.get_issue_events():
-            if e.event not in ("review_requested", "review_request_removed"):
-                continue
 
-            yield {
-                "when": e.created_at,
-                "what": e.event.upper(),
-                "who": e.requested_reviewer.login
+    QUERY = """
+query($org:String!, $repo:String!, $pr:Int!) {
+  repository(owner: $org, name: $repo) {
+    pullRequest(number: $pr) {
+      timelineItems(first:100) {
+        nodes {
+          __typename
+          ...on PullRequestReview {
+            author {
+              login
             }
-
-        for r in pr.get_reviews():
-            now = datetime.datetime.now()
-            yield {
-                "when": r.submitted_at if r.state != 'PENDING' else now,
-                "what": r.state,
-                "who": r.user.login
+            state
+            createdAt
+          }
+          ...on ReviewRequestedEvent {
+            requestedReviewer {
+              ...on User {
+                login
+              }
             }
-
-    return sorted(inner(pr), key=lambda x: x["when"])
-
-def get_responsible(pr):
+            createdAt
+          }
+          ...on ReviewRequestRemovedEvent {
+            requestedReviewer {
+              ...on User {
+                login
+              }
+            }
+            createdAt
+          }
+        }
+      }
+    }
+  }
+}
+
+    """
+    for e in enarxbot.graphql(QUERY, org=repo.organization.name, repo=repo.name, pr=pr.number)['repository']['pullRequest']['timelineItems']['nodes']:
+        if e['__typename'] not in ("PullRequestReview", "ReviewRequestedEvent", "ReviewRequestRemovedEvent"):
+            continue
+        yield {
+            "when": e['createdAt'],
+            "what": e['state'] if e['__typename'] == "PullRequestReview" else e['__typename'],
+            "who": e['author']['login'] if e['__typename'] == "PullRequestReview" else e['requestedReviewer']['login']
```

Comment:
This can be done efficiently with a graphql alias: https://graphql.org/learn/queries/#aliases